SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -786,87 +789,56 @@ ERPCResult USpatialSender::SendRPCInternal(UObject* TargetObject, UFunction* Fun
 			UE_LOG(LogSpatialSender, Verbose, TEXT("Sending unreliable command request (entity: %lld, component: %d, function: %s)"),
 				EntityId, CommandRequest.component_id, *Function->GetName());
 		}
-
-		return ERPCResult::Success;
 	}
-	case ERPCType::NetMulticast:
-	case ERPCType::ClientReliable:
-	case ERPCType::ServerReliable:
-	case ERPCType::ClientUnreliable:
-	case ERPCType::ServerUnreliable:
+	else
 	{
-		FUnrealObjectRef TargetObjectRef = PackageMap->GetUnrealObjectRefFromObject(TargetObject);
-		if (TargetObjectRef == FUnrealObjectRef::UNRESOLVED_OBJECT_REF)
-		{
-			return ERPCResult::UnresolvedTargetObject;
-		}
+		Worker_EntityId EntityId = TargetObjectRef.Entity;
+		check(EntityId != SpatialConstants::INVALID_ENTITY_ID);
 
-		if (SpatialGDKSettings->UseRPCRingBuffer() && RPCService != nullptr)
-		{
-			EPushRPCResult Result = RPCService->PushRPC(TargetObjectRef.Entity, RPCInfo.Type, Payload);
+		Worker_ComponentId ComponentId = SpatialConstants::RPCTypeToWorkerComponentIdLegacy(RPCInfo.Type);
 
-			if (Result == EPushRPCResult::Success)
-			{
-				FlushRPCService();
-			}
+		FWorkerComponentUpdate ComponentUpdate = CreateRPCEventUpdate(TargetObject, Payload, ComponentId, RPCInfo.Index);
 
+		Connection->SendComponentUpdate(EntityId, &ComponentUpdate);
+		Connection->MaybeFlush();
+	}
 #if !UE_BUILD_SHIPPING
-			if (Result == EPushRPCResult::Success || Result == EPushRPCResult::QueueOverflowed)
-			{
-				TrackRPC(Channel->Actor, Function, Payload, RPCInfo.Type);
-			}
+	TrackRPC(Channel->Actor, Function, Payload, RPCInfo.Type);
```

Comment:
What are the rules of when we track and rpc?

Reply:
Good point! It's missing for `OutgoingOnCreateEntityRPCs`, will put it back there.

Reply:
Resolving as deprecated.